js/api/BackgroundProcess.js

```diff
@@ -11,55 +13,156 @@ export const BackgroundGeolocation = Platform.OS === 'ios' ?
+        stopDetectionDelay: 0,
+        stopTimeout: 2,
+
+        // Application config
+        debug: false,
+
+        // Android config
+        forceReloadOnLocationChange: false,
+        forceReloadOnMotionChange: false,
+        forceReloadOnGeofence: false,
+        stopOnTerminate: true,
+        startOnBoot: true,
+
+        // iOS config
+        stationaryRadius: 200, // restart tracking after the user has moved 200m
+        useSignificantChangesOnly: true,
+        disableMotionActivityUpdates: true,
+      };
+    } else {
+      // High Accuracy Mode
+      console.log('Configuring Geolocation: High Accuracy Mode');
+      config = {
+        // License validations
+        orderId: Settings.licenses.BackgroundGeolocation.bundleId,
+        license: Settings.licenses.BackgroundGeolocation.key,
+
+        // Geolocation config
+        desiredAccuracy: 10,
+        distanceFilter: 50,
+        locationUpdateInterval: 10000,
+        fastestLocationUpdateInterval: 10000,
+
+        // Activity Recognition config
+        minimumActivityRecognitionConfidence: 80,
+        activityRecognitionInterval: 15000,
+        stopDetectionDelay: 1,
+        stopTimeout: 5,
+
+        // Application config
+        debug: false,
+
+        // Android config
+        forceReloadOnLocationChange: false,
+        forceReloadOnMotionChange: false,
+        forceReloadOnGeofence: false,
+        stopOnTerminate: true,
+        startOnBoot: true,
+
+        // iOS config
+        stationaryRadius: 100, // restart tracking after the user has moved 100m
+        useSignificantChangesOnly: false,
+        disableMotionActivityUpdates: false,
+      }
+    }
+    if (options.isInitial) {
+      BackgroundGeolocation.configure(config, callback);
+    } else {
+      BackgroundGeolocation.setConfig(config);
+      if (callback) callback();
+    }
   } catch (e) {
     console.error(e)
   }
 }
 
+/**
+ * Stops any running background services and initializes a new process.
+ */
 export function initializeGeolocationService() {
-  configureGeolocationService((state) => {
+  BackgroundGeolocation.stop();
+  configureGeolocationService({isInitial: true}, (state) => {
+
     BackgroundGeolocation.on('error', function(error) {
-      // printTimelog('error')
+      printTimelog('error');
       console.log(error.type + " Error: " + error.code)
-    })
+    });
 
     // Create initial geofence hooks.
-    setupGeofences()
+    setupGeofences(()=>{
+      BackgroundGeolocation.start((state) => {
+        Store.backgroundServiceState = 'started';
+        console.info('Geolocation tracking started.');
+      });
+    });
   })
 }
 
+/**
+ * Configures the background service to use less energy when the app is not on the foreground.
+ * Ignores 'inactive' state. (transitioning, using phone, etc.)
+ * @param  {string} state The new state the app is transitioning into: 'active', 'inactive', 'background'
+ */
+export function handleAppStateChange(state) {
```

Comment:
Please explain why on L142 `BackgroundGeolocation.start((state) => {` and L155 is `BackgroundGeolocation.startGeofences((state) => {`.